### PR TITLE
Issue 786030 revise RDTSC logic to consider CPU speeds taken from /proc/cpuinfo

### PR DESCRIPTION
**Pull request by rosenbaumalex**
1. Read the CPU's Hz (based on /proc/cpuinfo Mhz report)
2. Provide the MAX and MIN values, which might be the case if core are running at power control states
3. Report CPu speed on libvma bringup
4. Update time_measurements_test

a 'side effect' of this commit removes the potential blocking in 'usleep(1000)' call form get_tcs_rate_per_second().
1. Increase accuracy in the calibrate RDTSC with CPU speed - new logic is not disturbed by system context switches
2. This resolves coverify issue: "Program Hangs"

Note: to allow this change I had to move several of the utilities from "src/vma/util" to generic static library in "src/utils" (similar to "vlogger")

### CODE REVIEW

**Comment by rosenbaumalex** [comment]
from quick testing it seems to have solved: https://github.com/Mellanox/libvma/issues/98

**Comment by Di0gen** [comment]
bot:retest

**Comment by Di0gen** [comment]
bot:retest

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/319/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/337/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/338/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/339/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/320/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/321/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/341/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/323/ for details (Mellanox internal link).

File: src/utils/rdtsc.cpp [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+#include <stdio.h>
+#include <sys/param.h> // for MAX & MIN
+#include "rdtsc.h"
+
+
+bool get_cpu_hz(double &hz_min, double &hz_max)
+{
+	FILE* f;
+	char buf[256];
+
```
Comment (igor-ivanov) [inline src/utils/rdtsc.cpp]: @rosenbaumalex do you see any advantage to calculate it min/max in every call()? May be we could calculate once and return stored values?

Comment (rosenbaumalex) [inline src/utils/rdtsc.cpp]: I think not in this level.
The `rdtsc.h:get_tsc_rate_per_second()` will cache the result and never call it again.

In cases where CPU is not configured for high performance (some 'other' power state), then CPU core speed can vary. In these cases we'll get MAX != MIN cpu Mhz speed.
In these cases speed can increase/decrease other time. If we will want to re-check we should have a way to read current value and not only on startup of process. 
At this point I don't see this happening but don't want to block this option.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/328/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/346/ for details (Mellanox internal link).

**Comment by OphirMunk** [comment]
Original issue name was: "libvma time is not monotonically increasing". Is this PR still addresses the original issue?

**Comment by rosenbaumalex** [comment]
@OphirMunk, yes. Seems that this solves the original issue reported about time moving back in some cases. 
But I think the current name better reflects the changes done in the code.

**Comment by OphirMunk** [comment]
Getting the following log output while running this PR.
```
Time:  9728.236 Pid: 25336 Tid: 25336 VMA FINE: bpool[0x1667620]:189:free_bpool_resources() count 200000, missing 0
Time:  9929.734 Pid: 25336 Tid: 25336 VMA FINER: ENTER: write(fd=7, nbytes=12)
Time:  9984.406 Pid: 25336 Tid: 25336 VMA DEBUG: vma_stats_instance_remove_bpool_block:566: Remove bpool local=0x16676d0
Time:  9755.605 Pid: 25336 Tid: 25336 VMA FINE: bpool[0x1667620]:219:free_bpool_resources() done
Time:  9755.612 Pid: 25336 Tid: 25336 VMA FINE: bpool[0x16671c0]:189:free_bpool_resources() count 200000, missing 0
Time:  9984.461 Pid: 25336 Tid: 25336 VMA FINER: ENTER: write(fd=7, nbytes=12)
Time: 10033.694 Pid: 25336 Tid: 25336 VMA DEBUG: vma_stats_instance_remove_bpool_block:566: Remove bpool local=0x1667270
Time:  9780.239 Pid: 25336 Tid: 25336 VMA FINE: bpool[0x16671c0]:219:free_bpool_resources() done
Time: 10033.739 Pid: 25336 Tid: 25336 VMA DEBUG: nl_wrapper:162:~netlink_wrapper() ---> netlink_route_listener DTOR (LIBNL1)
Time: 10033.772 Pid: 25336 Tid: 25336 VMA DEBUG: ENTER: close(fd=16)
```

**Comment by rosenbaumalex** [comment]
bot:retest

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/350/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/332/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/340/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/358/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/356/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/374/ for details (Mellanox internal link).

**Comment by OphirMunk** [comment]
@rosenbaumalex You need to rebase this PR on latest master. Then issue push --force

**Comment by rosenbaumalex** [comment]
@OphirMunk pushed rebased code... jenkins working on it

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/359/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/377/ for details (Mellanox internal link).

**Pull request merged**